### PR TITLE
Removing ``isEmpty()`` comment from hamming analyzer

### LINKED ISSUE #94: Remove comment to use `isEmpty()` from Hamming analyzer
So I'm getting this recommendation on the hamming practice exercise.

image

And checking the solutions that are mostly like the one in as [reference](https://github.com/exercism/java/blob/main/exercises/practice/hamming/.meta/src/reference/java/Hamming.java) in the exercise definition, I'm not understanding how this helps improve the code.

Because if both strands are empty the code works properly and the count is 0, there is even a [test](https://github.com/exercism/java/blob/1a73c1e6c6de4b6224e25e996e653f7949e54bf3/exercises/practice/hamming/src/test/java/HammingTest.java#L10) that checks that

### PR DESCRIPTION
**Pull request by manumafe98**
Fixes #94

The goal is to remove an old analyzer comment that after an exercise reformat is not useful anymore.

I guess is should open a PR to remove [this](https://github.com/exercism/website-copy/blob/main/analyzer-comments/java/hamming/should_use_string_is_empty.md) from the website-copy right?

### CODE REVIEW

**Comment by manumafe98** [comment]
Sidenote: the gradlew.bat file changed alone, but it seems like is the same, I tried restoring the changes but does not seem to work for some reason.

File: src/main/java/analyzer/exercises/hamming/HammingAnalyzer.java [hunk 1]
```diff
@@ -57,10 +57,6 @@ public void analyze(Solution solution, Analysis analysis) {
             analysis.addComment(new CalculateDistanceInConstructor());
         }
 
-        if (!walker.usesStringIsEmpty()) {
```
Comment (sanderploegsma) [inline src/main/java/analyzer/exercises/hamming/HammingAnalyzer.java:60]: You can also remove the `usesStringIsEmpty()` method (and corresponding fields and helper methods if there are any) from the `HammingWalker` class.

**Review by sanderploegsma — CHANGES_REQUESTED** [review]

**Comment by sanderploegsma** [comment]
> Sidenote: the gradlew.bat file changed alone, but it seems like is the same, I tried restoring the changes but does not seem to work for some reason.

That's fine, I think the line endings were automatically converted by Git. No need to revert it!

**Comment by sanderploegsma** [comment]
> I guess is should open a PR to remove this from the website-copy right?

Not strictly required, I believe there are some more templates there that are no longer being used. We need to do some housekeeping there anyway.

You are welcome to open a PR to remove the template though. If you do, we should make sure it's not merged before this PR.

File: src/test/java/analyzer/exercises/hamming/HammingAnalyzerTest.java [hunk 2]
```diff
@@ -25,7 +25,6 @@ private static Stream<Arguments> testCases() {
                 Arguments.of("MustUseCharAtOrCodePointAt.java.txt", new Comment[]{new MustUseStringCharAtOrCodePointAt()}),
                 Arguments.of("NestedValidation.java.txt", new Comment[]{new CalculateDistanceInConstructor()}),
                 Arguments.of("NestedCalculation.java.txt", new Comment[0]),
-                Arguments.of("ShouldUseStringIsEmpty.java.txt", new Comment[]{new ShouldUseStringIsEmpty()}),
```
[the remark this answers is not in the document]
Comment (manumafe98) [reply src/test/java/analyzer/exercises/hamming/HammingAnalyzerTest.java:28]: Done!

**Comment by manumafe98** [comment]
It's normal that the pipeline is failing? there were two steps, a docker one, and the coverage one, or i'm missing something else? Thanks

**Comment by sanderploegsma** [comment]
> It's normal that the pipeline is failing? there were two steps, a docker one, and the coverage one, or i'm missing something else? Thanks

The `tests` folder contains a couple of smoke tests for each exercise. It looks like the one in `tests/hamming` is expecting to receive the removed comment, see `tests/hamming/expected_analysis.json`. You can just update that file to no longer expect that comment to be present, and the smoke tests should turn green again.

**Comment by sanderploegsma** [comment]
As for the test coverage decreasing, that is expected since you removed some lines that were covered, so the overall coverage of the affected files drops with it. No need to worry about that.

**Comment by manumafe98** [comment]
> > It's normal that the pipeline is failing? there were two steps, a docker one, and the coverage one, or i'm missing something else? Thanks
> 
> The `tests` folder contains a couple of smoke tests for each exercise. It looks like the one in `tests/hamming` is expecting to receive the removed comment, see `tests/hamming/expected_analysis.json`. You can just update that file to no longer expect that comment to be present, and the smoke tests should turn green again.

Done!

**Review by sanderploegsma — APPROVED** [review]
Great stuff!

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**